### DISCUSSION TITLE
M5Stick C Plus haspmota support?

### DISCUSSION THREAD

**Discussion by miroc99** [arendst/Tasmota#17700] [Q&A]
I have a M5stick C Plus and I flash tasmota follow [this guide](https://templates.blakadder.com/m5stack_stickc.html), which is for the old version M5Stick C with a small screen. All funtion work great except the dispslay, with `tamota_demo.tapp` it show some of the content but not at right position and mirrored. I don't know it was some setting problem or just this device not supported yet. 
I wonder if it is possible to change some config about screen resolution to fix it.
![](https://upload.cc/i1/2023/01/14/rZ5IH1.jpeg)

**Comment by Jason2866** [comment]
The display.ini needs to be adopted to the used display in your device.
Please provide a link where it is described which display is used.
Sadly M5 has so many devices with similar names and different components inside.

**Reply by Jason2866** [reply]
Have you this display.ini in use?
```
:H,ST7735S,160,80,16,SPI,1,*,*,*,*,*,*,*,16
:I
01,80
11,E0
B1,3,05,3C,3C
B2,3,05,3C,3C
B3,6,05,3C,3C,05,3C,3C
B4,1,07
C0,3,A4,04,84
C1,1,C5
C2,2,0D,00
C3,2,8D,6A
C4,2,8D,EE
C5,1,21
FC,1,80
F0,1,11
D6,1,CB
20,0
36,1,C8
3A,1,05
2A,4,00,02,00,81
2B,4,00,01,00,A0
E0,10,0D,0C,0C,0E,0E,00,00,00,00,09,23,31,00,0C,03,1A
E1,10,0A,05,06,07,08,01,00,00,00,05,20,2E,00,0A,01,1A
13,A0
29,E0
:o,28
:O,29
:A,2A,2B,2C,16
:R,36
:0,68,01,1A,00
:1,CC,1A,01,01
:2,A8,01,1A,02
:3,08,1A,01,03
:i,21,20
:B,60,1
#
```

**Reply by miroc99** [reply]
I use the display.ini (ST7735S) at first, aftrer read your comment I found out M5stick C Plus use [1.14 inch, 135*240 Colorful TFT LCD, ST7789v2]([from this doc](https://docs.m5stack.com/en/core/m5stickc_plus)) , so I try this file that I grab from the tasmota repository.
Result is full black screen, only backlight is on. How can I modify it? 
```:H,ST7789,135,240,16,SPI,3,*,*,*,*,*,*,*,40
:S,2,1,3,0,80,30
:I
01,A0
11,A0
3A,81,55
36,81,00
21,80
13,80
29,A0
:o,28
:O,29
:A,2A,2B,2C
:R,36
:0,C0,35,28,00
:1,A0,28,34,01
:2,00,34,28,02
:3,60,28,35,03
:i,21,20
:TI2,38,32,23
:r,1
:B,30,5
#
```

**Comment by Jason2866** [comment]
It is described here https://tasmota.github.io/docs/Displays/#descriptor-file

**Comment by ase1590** [comment]
I am also interested in getting this display supported.

As mentioned above, the M5StickC-Plus uses the Sitronix ST7789v2 screen.
Docs for the screen can be found here: https://m5stack.oss-cn-shenzhen.aliyuncs.com/resource/docs/datasheet/unit/lcd/ST7789V2_SPEC_V1.0.pdf

**Reply by rafamiga** [reply]
This kinda works for me but use it at your own risk. Rotations are disabled (`:0` is copied to `:1`, `:2` and `:3`).

```
:H,ST7789,240,135,16,SPI,1,*,*,*,*,*,*,*,16
:I
20,0
36,1,A8
3A,1,05
B2,5,0C,0C,00,33,33
B7,1,35
BB,1,28
C0,1,0C
C2,2,01,FF
C3,1,0B
C4,1,20
C6,1,0F
D0,2,A4,A1
E0,0E,D0,00,02,07,0A,28,32,44,42,06,0E,12,14,17
E1,0E,D0,00,02,07,0A,28,31,42,40,06,0E,12,14,17
11,78
21,0
29,14
:o,28
:O,29
:A,2A,2B,2C,16
:R,36
:0,AA,2A,34,00
:1,AA,2A,34,00
:2,AA,2A,34,00
:3,AA,2A,34,00
:i,21,20
:B,60,1
#
```
